Update gcp documentation (#30929)
open-telemetry/opentelemetry-collector-contrib: exporter/googlecloudexporter/README.md

If set at Collector start time then exporters, regardless of protocol,
will or will not proxy traffic as defined by these environment variables.

### Monitored Resources

For metrics and logs, this exporter maps the OpenTelemetry Resource to a Google
Cloud [Logging\(https://cloud.google.com/logging/docs/api/v2/resource-list) or
[Monitoring](https://cloud.google.com/monitoring/api/resources) Monitored Resource.

The complete mapping logic can be found [here](https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/blob/main/internal/resourcemapping/resourcemapping.go).
That may be the most helpful reference if you want to map to a specific monitored
resource.

If running on GCP, using the GCP resource detector, as shown above, will populate
the resource attributes required to map to the appropriate monitored resource.

> **Note:** If you are not running on GCP, you still need to choose a [GCP zone or

Reviewer: Maybe it would make sense to put these sections under another set of headers? One for running on GCP, the other for not running on GCP? Just thinking of ways to try to make it more obvious.

Author: done

region](https://cloud.google.com/compute/docs/regions-zones) to send telemetry to

> by setting cloud.availability_zone or cloud.region.

Reviewer: ```suggestion
by setting `cloud.availability_zone` or `cloud.region`.
```
Nit: formatting

> If running on GCP, using the GCP resource detector, as shown above, will populate
> the resource attributes required to map to the appropriate monitored resource.
> 
> **Note:** If you are not running on GCP, you still need to choose a [GCP zone or
> region](https://cloud.google.com/compute/docs/regions-zones) to send telemetry to
> by setting `cloud.availability_zone` or `cloud.region`.

Author: ```suggestion
#### On GCP

If running on GCP, using the GCP resource detector, as shown above, will populate
the resource attributes required to map to the appropriate monitored resource.

#### Off GCP
If you are not running on GCP, you still need to choose a [GCP zone or
region](https://cloud.google.com/compute/docs/regions-zones) to send telemetry to
by setting `cloud.availability_zone` or `cloud.region`. In addition, you should use the detector associated with other cloud providers, if applicable.
```

> If running on Kubernetes, it is recommended to additionally set k8s.pod.name,

Reviewer: ```suggestion
If running on Kubernetes, it is recommended to additionally set `k8s.pod.name`,
```
Nit: formatting

If set at Collector start time then exporters, regardless of protocol,
will or will not proxy traffic as defined by these environment variables.

### Monitored Resources

For metrics and logs, this exporter maps the OpenTelemetry Resource to a Google
Cloud [Logging\(https://cloud.google.com/logging/docs/api/v2/resource-list) or
[Monitoring](https://cloud.google.com/monitoring/api/resources) Monitored Resource.

The complete mapping logic can be found [here](https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/blob/main/internal/resourcemapping/resourcemapping.go).
That may be the most helpful reference if you want to map to a specific monitored
resource.

#### On GCP

If running on GCP, using the GCP resource detector, as shown above, will populate
the resource attributes required to map to the appropriate monitored resource.

#### Off GCP

> If you are not running on GCP, you still need to choose a [GCP zone or

Reviewer: ```suggestion

If you are not running on GCP, you still need to choose a [GCP zone or
```
Nit: whitespace consistency

      - delete_key(attributes, "instrumentation_version")
```

**Note** It is not recommended to use these transformations with the googlecloud

> exporter in a logging pipeline.

Reviewer: From the [referenced issue](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/27803):
```
 We should make clear that the documentation does not apply to logging or trace, and that it should (probably) only be used with the prometheus receiver.
```
Should we expand this message in the README to include this, or did I misunderstand the context?

Author: This particular part of the documentation should be not recommended for logging or trace.  Added trace to the statement.

> exporter in a logging pipeline.

Author: ```suggestion
exporter in a logging or trace pipeline.
```
